locations/spiders/screwfix.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+import scrapy
+from locations.items import GeojsonPointItem
+
+class ScrewfixSpider(scrapy.Spider):
+	name = "screwfix"
+	allowed_domains = ["www.screwfix.com"]
+	download_delay = 0.1
+	start_urls = (
+		"https://www.screwfix.com/jsp/tradeCounter/tradeCounterAllStoresPage.jsp",
+	)
+
+	def parse(self, response):
+		storeurls = response.xpath(".//ul[@class='store-name']//a/@href").extract()
+		storeurls = [response.urljoin(i) for i in storeurls]
+		for i in range(len(storeurls)):
+			yield scrapy.Request(storeurls[i], callback=self.parse_location)
```

Comment:
Instead of iterating over the indexes of the list, just index over the list of joined URLs:

```suggestion
		for storeurl in response.xpath(".//ul[@class='store-name']//a/@href").extract():
			yield scrapy.Request(response.urljoin(storeurl), callback=self.parse_location)
```